File: acmp.ru/[0701] - [0800]/[0798] Шахматная доска - 2.cpp
#include <iostream>
using namespace std;

int main() {
    int n, m, i, j, c;
    cin >> n >> m >> i >> j >> c;

    if (n * m % 2 == 0)
        cout << "equal" << endl;
    else {
        if ((i + j) % 2 == 0)
            cout << (c ? "white" : "black") << endl;
        else
            cout << (c ? "black" : "white") << endl;
    }

    return 0;
}

File: acmp.ru/[0701] - [0800]/[0798] Шахматная доска - 2.py
n, m, i, j, c = [int(i) for i in input().split()]

if n * m % 2 == 0:
    print('equal')
else:
    if (i + j) % 2 == 0:
        print('white' if c else 'black')
    else:
        print('black' if c else 'white')
